actionview/lib/action_view/helpers/form_helper.rb

```diff
@@ -1879,6 +1879,7 @@ def submit_default_value
           end
 
           defaults = []
+          defaults << :"helpers.submit.#{object.model_name.i18n_key}.#{key}" if object.respond_to?(:model_name)
```

Comment:
What should I think about specification of `:as` option?
If `:as` option is given to `form_for` method, is it prior than actual model name in translation?
For example,

```ruby
class Blog::Post << ActiveRecord::Base
end
```

```erb
form_for @blog_post, as: :my_post do |f|
  f.submit
end
```

```yaml
en:
  helpers:
    submit:
      "blog/post":
        create: Post blog post
      my_post:
        create: Post my post
```

Then which value is expected `Post blog post` or `Post my post`?